exercises/luhn/canonical-data.json

```diff
@@ -105,6 +105,14 @@
         "value": "091"
       },
       "expected": true
+    },
+    {
+      "description": "strings with non-digits is invalid",
```

Comment:
How is this new case different from `"valid strings with a non-digit included become invalid"`?

Reply:
```go
for i, char := range str {
    n := int(char - '0')
    if (len(str)-i)%2 == 0 {
        n *= 2
        if n >= 10 {
            n -= 9
        }
    }
    sum += n
}
```

The code above will pass current tests, but not this one.

So we have to check non-digits by adding:

```go
if n > 9 || n < 0 {
    return false
}
```

Reply:
Thanks for the explanation.  I can see now how `"valid strings with a non-digit included become invalid"` passes without the new test case.

Other reviewers may ask that your new test case be moved to an earlier location in the JSON file.  I'll leave that open to others to address.

@idealhack thanks for your time in improving this exercise! :+1: